api/appconfig.go

```diff
@@ -111,5 +128,61 @@ func fetchAppConfig(url string, deploymentRequest NaisDeploymentRequest) (naisAp
 		return NaisAppConfig{}, err
 	}
 
+	validationErrors := validateAppConfig(appConfig);
+	if len(validationErrors.Errors) != 0 {
+		glog.Error("Invalid appconfig: ", validationErrors.Error())
+		return NaisAppConfig{}, &validationErrors
+	}
+
 	return appConfig, nil
 }
+
+func validateAppConfig(appConfig NaisAppConfig) ValidationErrors {
```

Comment:
Would extracting each validation into its own function lead to better tests and readability?

Reply:
fixing